### PR TITLE
Improve developer docs on building/writing documentation

### PR DESCRIPTION
**Pull request by bdice**
## Description
I noticed some older/inaccurate information in the developer docs about building and writing documentation. This fixes that info.

## Motivation and Context
Documenting good documentation practices, very meta.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:
- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,9 +24,9 @@ In order to make _freud_ as widely available as possible, we try to keep the num
 If you need a feature present in an external library, follow the following steps:
 
 1. Add to _freud_ itself if it's simple or if other modules would benefit:
-    * Example: Added simple tensor math for CubaticOrderParameter
+    * Example: Added simple tensor math for Cubatic order parameter.
```
Comment (vyasr) [inline CONTRIBUTING.md]:
```suggestion
    * Example: Added simple tensor math for cubatic order parameter.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -24,9 +24,9 @@ In order to make _freud_ as widely available as possible, we try to keep the num
...
 2. Add via submodule if the code exists externally:
-    * Example: _fsph_
+    * Example: _fsph_ for spherical harmonics in the Steinhardt order parameter.
```
Comment (vyasr) [inline CONTRIBUTING.md]:
```suggestion
    * Example: _fsph_ for spherical harmonics.
```

**Review by vyasr — APPROVED** [review]
Since this is super developer side I'm going to go ahead and make the proposed changes and merge when ready, @klywang there are other PRs open for you to review though :)

File: doc/source/reference/development/howtoadd.rst [hunk 3]
```diff
@@ -40,6 +40,8 @@ Please refer to `Clang Format 6 <http://releases.llvm.org/6.0.1/tools/clang/docs
 When in doubt, run :code:`clang-format -style=file FILE_WITH_YOUR_CODE` in the top directory of the **freud** repository.
 If installing :code:`clang-format` is not a viable option, the :code:`check-style` step of continuous integration (CI) contains the information on the correctness of the style.
 
+Doxygen docstrings should be used for classes, functions, etc.
```
Comment (klywang) [inline doc/source/reference/development/howtoadd.rst:43]: Should we mention that developers should ignore PEP E501 errors for these docstrings?

**Pull request merged**